From 6565a713d360acc291e47f834e85f81c2630c281 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 6 Sep 2025 00:16:05 +0800
Subject: [PATCH 86/96] ia64: sal: Dirtily fix build on new gcc

When -Werror is enabled, it does not allow arbitrary function pointer
casts, so return a dummy struct instead.

arch/ia64/kernel/sal.c:43:29: error: cast between incompatible function types from 'long int (*)(void)' to 'struct ia64_sal_retval (*)(u64, ...)' {aka 'struct ia64_sal_retval (*)(long long unsigned int, ...)'} [-Werror=cast-function-type]
   43 | ia64_sal_handler ia64_sal = (ia64_sal_handler) default_handler;
      |                             ^

Signed-off-by: Henry Chen <chenx97@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/ia64/kernel/sal.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/ia64/kernel/sal.c b/arch/ia64/kernel/sal.c
index e4f0705c0282..13139db54942 100644
--- a/arch/ia64/kernel/sal.c
+++ b/arch/ia64/kernel/sal.c
@@ -34,10 +34,11 @@ static struct {
 	void *gpval;	/* gp value to use */
 } pdesc;
 
-static long
-default_handler (void)
+static struct ia64_sal_retval
+default_handler (u64, ...)
 {
-	return -1;
+	struct ia64_sal_retval s = {.status = -1};
+	return s;
 }
 
 ia64_sal_handler ia64_sal = (ia64_sal_handler) default_handler;
-- 
2.25.1

